### PR TITLE
New policy

### PR DESCRIPTION
**Pull request by wongband**
- Implemented auth policy to only allow users from the database to access routes
- Created a 403 error page
- Redirected users to 403 page if they access a route that they are not suppose to

### CODE REVIEW

File: src/FacultyDirectory/ClientApp/src/components/NavMenu.tsx [hunk 1]
```diff
@@ -8,12 +9,20 @@ interface State {
...
   let [name, setName] = useState('');
 
   useEffect(() => {
     const getName = async () => {
-      const results = await fetch('api/users/name').then(r => r.json());
-      setName(results.name);
+      const results = await fetch('api/users/name');
+      const response = await results;
+
+      if (response.status == 403) {
```
Comment (srkirkland) [inline src/FacultyDirectory/ClientApp/src/components/NavMenu.tsx:20]: `===`

File: src/FacultyDirectory/Helpers/AuthorizationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        public AuthorizationHandler(ApplicationDbContext dbContext)
+        {
+            this.dbContext = dbContext;
+        }
+
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                       AdminRequirement requirement)
+        {
+            var username = context.User.Identity.Name;
+            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
```
Comment (srkirkland) [inline src/FacultyDirectory/Helpers/AuthorizationHandler.cs:24]: better to use `.Any()` here.  Users table is small so it doesn't make a big difference, but no reason to get the first row back when you just want a true/false.  Ex: `dbContext.Users.Any(u => u.Username == username)`

File: src/FacultyDirectory/ClientApp/src/components/UserTable/Users.tsx [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
-	
+
   useEffect(() => {
     const getUsers = async () => {
-      const results = await fetch('api/users').then(r => r.json());
-      setUsers(results);
+      const results = await fetch('api/users');
+      const response = await results;
+
+      if (response.status == 403) {
```
Comment (srkirkland) [inline src/FacultyDirectory/ClientApp/src/components/UserTable/Users.tsx:16]: `===` here too

File: src/FacultyDirectory/Helpers/AuthorizationHandler.cs [hunk 4]
```diff
@@ -21,9 +21,9 @@ public AuthorizationHandler(ApplicationDbContext dbContext)
                                                        AdminRequirement requirement)
         {
             var username = context.User.Identity.Name;
-            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
+            var user = await dbContext.Users.AnyAsync(u => u.Username == username);
 
-            if (user != null)
+            if (user)
```
Comment (jSylvestre) [inline src/FacultyDirectory/Helpers/AuthorizationHandler.cs:26]: I'd either move the await into the if and toast the user value, or change the variable name to userExists or similar.

Comment (srkirkland) [reply src/FacultyDirectory/Helpers/AuthorizationHandler.cs:26]: yea probably worth changing the variable to something like `isUserAdmin`

**Pull request merged**